src/mapping/grouping/cluster_config.jl

```diff
@@ -0,0 +1,202 @@
+    Maximilian Gelbrecht et al 2021, Monte Carlo basin bifurcation analysis,
+    [2020 New J. Phys.22 03303](http://dx.doi.org/10.1088/1367-2630/ab7a05)
+[^Kriegel1996]: Ester, Kriegel, Sander and Xu: A Density-Based Algorithm for Discovering
+    Clusters in Large Spatial Databases with Noise
+[^Schubert2017]:
+    Schubert, Sander, Ester, Kriegel and Xu: DBSCAN Revisited, Revisited: Why and How You
+    Should (Still) Use DBSCAN
+"""
+struct GroupViaClustering{R<:Union{Real, String}, M<:Metric, F<:Function} <: GroupingConfig
+    clust_distance_metric::M
+    min_neighbors::Int
+    rescale_features::Bool
+    optimal_radius_method::R
+    num_attempts_radius::Int
+    silhouette_statistic::F
+    max_used_features::Int
+end
+
+function GroupViaClustering(;
+        clust_distance_metric=Euclidean(), min_neighbors = 10,
+        rescale_features=true, num_attempts_radius=100,
+        optimal_radius_method::Union{Real, String} = "silhouettes_optim",
+        silhouette_statistic = mean, max_used_features = 0,
+    )
+    return GroupViaClustering(
+        clust_distance_metric, min_neighbors,
+        rescale_features, optimal_radius_method,
+        num_attempts_radius, silhouette_statistic, max_used_features
+    )
+end
+
+#####################################################################################
+# API funtion (group features)
+#####################################################################################
+function group_features(features::Vector{<:AbstractVector}, config::GroupViaClustering)
+    (; min_neighbors, clust_distance_metric, rescale_features, optimal_radius_method,
+    num_attempts_radius, silhouette_statistic, max_used_features) = config
+
+    nfeats = length(features); dimfeats = length(features[1])
+    if dimfeats ≥ nfeats
+        throw(ArgumentError("""
+        Not enough features. The algorithm needs the number of features
+        $nfeats to be greater or equal than the number of dimensions $dimfeats
+        """))
+    end
+
+    if rescale_features
+        features = rescale_to_01(features)
+    end
+
+    # Obtain optimal radius
+    if optimal_radius_method isa String
+        if max_used_features == 0 || max_used_features > nfeats
+            features_for_optimal = features
+        else
+            features_for_optimal = StatsBase.sample(features, max_used_features; replace = false)
+        end
+        ϵ_optimal = optimal_radius_dbscan(
+            features_for_optimal, min_neighbors, clust_distance_metric,
+            optimal_radius_method, num_attempts_radius, silhouette_statistic
+        )
+    elseif optimal_radius_method isa Real
+        ϵ_optimal = optimal_radius_method
+    else
+        error("Specified optimal_radius_method is incorrect. Please specify the radius
+        directly as a Real number or the method to compute it as a String")
+    end
+    # Perform the DBSCAN and assign the result into labels
+    distances = pairwise(clust_distance_metric, features)
```

Comment:
The PR I am working on depends on these functions. For the MCBB algorithm you need to provide the distance matrix externally. I used a keyword argument `distances` wich is by default `nothing`

```
    if isnothing(distances)
        distances = pairwise(clust_distance_metric, features)
    end
```